bidict/_bidict.py

```diff
@@ -13,32 +13,31 @@ def _del(self, key):
         del self._bwd[val]
 
     def __delitem__(self, key):
+        """
+        Analogous to dict.__delitem__(), keeping bidirectionality intact.
+        """
         self._del(key)
 
     def __setitem__(self, key, val):
+        """
+        Analogous to dict.__setitem__(), keeping bidirectionality intact.
+
+        :raises ValueExistsException: if attempting to insert a mapping with a
+            non-unique value.
+        """
         self._put(key, val)
 
     def put(self, key, val):
         """
-        Alternative to using the setitem syntax to insert a mapping.
+        Alternative to using :attr:`__setitem__` to insert a mapping.
         """
         self._put(key, val)
```

Comment:
I think it may now be sensible for `put()` to be the "non-overwriting set", rather than an alias for `__setitem__`. For example,

    def put(self, key, val):
        if (key in self._fwd and self._fwd[key] != val) or (val in self._bwd and self._bwd[val] != key):
            raise ItemExistsException(something)
        self._put(key, val)

This closes off the case you struggle to explain in the Caveats docs, about how to insert an item without a risk of accidental overwrite on the key side as well as the value side. Meanwhile the `__setitem__` syntax on the forward and inverse bidicts allows overwrites on keys and values respectively, and `forceput()` allows overrides from both ends.